lib/api/metric.js

```diff
@@ -110,10 +93,26 @@ function send(metric, points, extra, callback){
 function send_all(metrics, callback){
     var now = parseInt(new Date().getTime() / 1000);
     for(var i = 0; i < metrics.length; ++i){
-        if(!Array.isArray(metrics[i].points)){
-            metrics[i].points = [now, metrics[i].points];
-        }
+        // Try to normalize `points`
+        // DEV: We need `points` to be an array of arrays regardless of what they give us
+        // Always wrap points in an array, this way we will get:
+        //   500 => [500]
+        //   [<timestamp>, 500] => [[<timestamp>, 500]]
+        points = [metrics[i].points];
```

Comment:
I think if you add a check `if(!Array.isArray(metrics[i].points))` for this line, you will cover all the examples I gave below.